### PR TITLE
New EFI updater

### PR DESCRIPTION
**Pull request by igor-stoppa**
Switch back to shell script EFI updater.

Signed-off-by: Igor Stoppa igor.stoppa@intel.com

### CODE REVIEW

File: meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Updates the EFI combo application
+BindsTo=update-triggers.target
+
+[Install]
+WantedBy=update-triggers.target
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/efi_combo_updater
```
Comment (pohly) [inline meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new]: Should this be ExecStart=/usr/bin/efi_combo_updater.sh? It's installed below with install -m 0744 ${WORKDIR}/efi_combo_updater.sh ${D}/usr/bin/

Are users expected to invoke this command? If not, then I wouldn't put it into the default shell search PATH.

**Review by pohly — CHANGES_REQUESTED** [review]
I suspect that this won't work as it stands (see comment about ExecStart), so -1 from me.

Some more information in the commit message also would be useful - _why_ is this change desirable? Does it fix known bugs? If so, which ones (short summary and bug number)?

File: meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new (hunk 1 above)
Comment (mythi) [inline meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new]: Should be

```
ExecStart=/usr/bin/efi_combo_updater.sh
```

File: meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+    file://efi_combo_updater.sh \
     file://ostroprojectorg.key \
 "
 
 do_install_append () {
+    rm -f ${D}/usr/bin/efi_combo_updater
+    install -m 0744 ${WORKDIR}/efi_combo_updater.sh ${D}/usr/bin/
+
+    rm -f ${D}/${systemd_system_unitdir}/efi-combo-trigger.service
+    install -m 0644 ${WORKDIR}/efi-combo-trigger.service.new ${D}/${systemd_system_unitdir}/efi-combo-trigger.service
```
Comment (mythi) [inline meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend:14]: I'm OK with .new approach but another alternative would have been just

```
sed -i -e 's/^\(ExecStart.*\)/\1\.sh/' ${D}/${systemd_system_unitdir}/efi-combo-trigger.service
```

**Comment by igor-stoppa** [comment]
test this please

File: meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new (hunk 1 above)
Comment (igor-stoppa) [reply meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new]: I wasn't expecting such eagerness in the review :-)
The patch is not final, but I wanted to start checking if something fails in CI.

Comment (igor-stoppa) [reply meta-ostro-xt/recipes-swupd/swupd-client/swupd-client/efi-combo-trigger.service.new]: oops, yes

File: meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend (hunk 2 above)
Comment (igor-stoppa) [reply meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend:14]: I consider this entire patchset to be temporary for the XT release and I expect the hack to be dropped once we can use again meta-ostro from ostro-os.

In the meanwhile I prefer to have more self-contained changes.

Comment (mythi) [reply meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend:14]: Makes sense, thanks Igor

File: meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend [hunk 3]
```diff
@@ -1,10 +1,17 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
 SRC_URI_append = " \
+    file://efi_combo_updater.sh \
```
Comment (mythi) [inline meta-ostro-xt/recipes-swupd/swupd-client/swupd-client_%.bbappend:4]: do_install fails with

```
07:55:06 | install: cannot stat '/srv/jenkins/workspace/builder-slot-0/build/tmp-glibc/work/corei7-64-ostro-linux/swupd-client/3.6.0+gitAUTOINC+f4000c5b22-r0/efi-combo-trigger.service.new': No such file or directory
```

Not sure why fetcher works correctly but looks like `file://efi-combo-trigger.service.new` is missing

**Comment by igor-stoppa** [comment]
test this please

**Comment by igor-stoppa** [comment]
test this please

**Pull request merged**